readium/shared/src/main/java/org/readium/r2/shared/util/Language.kt

```diff
@@ -6,42 +6,49 @@
 
 package org.readium.r2.shared.util
 
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
 import java.util.*
 
 /**
  * Represents a language with its region.
  *
  * @param code BCP-47 language code
- * @param locale Java [Locale]
  */
-class Language private constructor(val code: String, val locale: Locale) {
+@Serializable(with = Language.Serializer::class)
+data class Language(val code: String) {
 
     /**
      * Creates a [Language] from a Java [Locale].
      */
-    constructor(locale: Locale) : this(code = locale.toLanguageTag(), locale = locale)
-
-    companion object {
-        /**
-         * Creates a [Language] from a BCP-47 language code.
-         */
-        operator fun invoke(code: String): Language {
-            val fixedCode = code.replace("_", "-")
-            return Language(code = fixedCode, locale = Locale.forLanguageTag(code))
-        }
-    }
+    constructor(locale: Locale) : this(code = locale.toLanguageTag())
+
+    val normalizedCode by lazy { code.replace("_", "-") }
+
```

Comment:
I think equality would fail unless `Bcp47Code` is a data class or implement equality, in which case it shifts the problem to another class.

I will make `Language` a regular class and implement equality.